addons/boa/tasks.py

```diff
@@ -0,0 +1,198 @@
+        return BoaErrorCode.UNKNOWN
+    logger.info('Query successfully submitted.')
+    logger.debug(f'Waiting for job to finish: job_id=[{str(boa_job.id)}] ...')
+    while boa_job.is_running():
+        logger.debug(f'Boa job still running, waiting 10s: job_id=[{str(boa_job.id)}] ...')
+        boa_job.refresh()
+        await asyncio.sleep(10)
+    if boa_job.compiler_status is CompilerStatus.ERROR:
+        client.close()
+        message = f'Boa job failed with compile error: job_id=[{str(boa_job.id)}]!'
+        await sync_to_async(handle_boa_error)(message, BoaErrorCode.QUERY_ERROR, user.username, user.fullname, project_url,
+                                              file_full_path, query_file_name=query_file_name, job_id=boa_job.id)
+        return BoaErrorCode.QUERY_ERROR
+    elif boa_job.exec_status is ExecutionStatus.ERROR:
+        client.close()
+        message = f'Boa job failed with execution error: job_id=[{str(boa_job.id)}]!'
+        await sync_to_async(handle_boa_error)(message, BoaErrorCode.QUERY_ERROR, user.username, user.fullname, project_url,
+                                              file_full_path, query_file_name=query_file_name, job_id=boa_job.id)
+        return BoaErrorCode.QUERY_ERROR
+    else:
+        try:
+            boa_job_output = boa_job.output()
+        except BoaException:
+            client.close()
+            message = f'Boa job output is not available: job_id=[{str(boa_job.id)}]!'
+            await sync_to_async(handle_boa_error)(message, BoaErrorCode.OUTPUT_ERROR, user.username,
+                                                  user.fullname, project_url, file_full_path,
+                                                  query_file_name=query_file_name, job_id=boa_job.id)
+            return BoaErrorCode.OUTPUT_ERROR
+        logger.info('Boa job finished.')
+        logger.debug(f'Boa job output: job_id=[{str(boa_job.id)}]\n########\n{boa_job_output}\n########')
+        client.close()
+        logger.debug('Boa client closed.')
+
+    logger.debug(f'Uploading Boa query output to OSF: name=[{output_file_name}], upload_url=[{output_upload_url}] ...')
+    try:
+        upload_request = request.Request(f'{output_upload_url}&name={output_file_name}')
+        upload_request.method = 'PUT'
+        upload_request.data = ensure_bytes(boa_job_output)
+        upload_request.add_header('Cookie', f'{osf_settings.COOKIE_NAME}={cookie_value}')
+        request.urlopen(upload_request)
+    except (ValueError, HTTPError, URLError) as e:
+        message = f'Failed to upload query output file to OSF: ' \
+                  f'name=[{output_file_name}], user=[{user_guid}], url=[{output_upload_url}]!'
+        error_code = BoaErrorCode.UPLOAD_ERROR_OTHER
+        if isinstance(e, HTTPError):
+            message += f', http_error=[{e.code}: {e.reason}]'
+            if e.code == 409:
+                error_code = BoaErrorCode.UPLOAD_ERROR_CONFLICT
```

Comment:
Are naming conflicts guaranteed when running a job more than once? Would it be feasible to upload a new version, in that case?